src/main/scala/mesosphere/marathon/state/PortDefinition.scala

```diff
@@ -14,7 +14,7 @@ case class PortDefinition(
 
 object PortDefinition {
   implicit val portDefinitionValidator = validator[PortDefinition] { portDefinition =>
-    portDefinition.protocol is oneOf(DiscoveryInfo.Port.AllowedProtocols)
+    portDefinition.protocol.split(',').toIterable is every(oneOf(DiscoveryInfo.Port.AllowedProtocols))
```

Comment:
Oh sorry, missed that. I was talking about "we can't reasonably change how
portDefs work in apps". Agree that we should allow a free form string for
pods.
On Thu, Sep 1, 2016 at 11:57 AM Johannes Unterstein <
notifications@github.com> wrote:

> In src/main/scala/mesosphere/marathon/state/PortDefinition.scala
> <https://github.com/mesosphere/marathon/pull/4284#discussion_r77233171>:
>
> > @@ -14,7 +14,7 @@ case class PortDefinition(
> >
> >  object PortDefinition {
> >    implicit val portDefinitionValidator = validator[PortDefinition] { portDefinition =>
> > -    portDefinition.protocol is oneOf(DiscoveryInfo.Port.AllowedProtocols)
> > +    portDefinition.protocol.split(',').toIterable is every(oneOf(DiscoveryInfo.Port.AllowedProtocols))
>
> In portMappings we accept tcp,udp as string, see
> https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/state/Container.scala#L103
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4284/files/a1c71b58309775a3ca78fc02af2fe987bc5ee603#r77233171>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAweZiv9jjfgPdXpvoRSaoEnQvdHzyJkks5qlx_3gaJpZM4Jysrn>
> .
>